frontend/src/components/shelter_sustainability/billOfQuantities/graphTree.vue

```diff
@@ -35,83 +51,368 @@ export default class GraphTree extends Vue {
+          const name = `${matched?.form}`;
+          return {
+            name,
+            value: [child[key], unitName],
+            itemStyle: {
+              color: currentColors[childIndex + 1],
+            },
+          } as datatree;
+        }),
+      } as datatree;
+    });
+  }
+  public get sankeyData(): sankeyData[] {
+    const result: Record<string, sankeyData> = {};
+    const localMaterialMap = this.materialMap;
+    const key = this.selectedField as MaterialTreeKey;
+    const unitName = this.unitName as string;
+
+    this.itemsWithoutTotal.forEach((item: MaterialTree) => {
+      const currentMaterialId = item.materialId ?? "";
+      const currentColors = this.materialColorsId[currentMaterialId] ?? [];
+
+      result[currentMaterialId] = {
+        name: currentMaterialId,
+        value: [item[key], unitName] as number[],
+        itemStyle: {
+          color: currentColors[0],
+        },
+      };
+      item?.children?.forEach((child: MaterialTree, childIndex) => {
+        const matched = localMaterialMap[child.formId as string] ?? {};
+        // const name = `${matched?.material} — ${matched?.form}`;
+        const name = `${matched?.form}`;
+        const currentValue = child[key] as number;
+        const previousValue = result[name]?.value?.[0];
+        result[name] = {
+          name,
+          value: [
+            previousValue ? currentValue + previousValue : currentValue,
+            unitName,
+          ] as number[],
+          itemStyle: {
+            color: currentColors[childIndex + 1],
+          },
+        };
+      });
+    });
+    return Object.values(result);
+  }
+
+  public get sankeyLinks(): sankeyLinks[] {
+    const result: sankeyLinks[] = [];
+    const localMaterialMap = this.materialMap;
+    const key = this.selectedField as MaterialTreeKey;
+    const unitName = this.unitName as string;
+
+    let source: string;
+    this.itemsWithoutTotal.forEach((item: MaterialTree) => {
+      source = item.materialId ?? "";
+      item?.children?.forEach((child: MaterialTree) => {
+        const matched = localMaterialMap[child.formId as string] ?? {};
+        // const name = `${matched?.material} — ${matched?.form}`;
+        const name = `${matched?.form}`;
+        result.push({
+          source,
+          target: name,
+          value: [child[key], unitName] as number[],
+        });
+      });
+    });
+    return result;
   }
 
   public get option(): EChartsOption {
     if (!this.selectedField || !this.unitName) {
       return {};
     }
+    if (this.graphType === "sunburst") {
+      return this.optionSunburst;
+    }
+    if (this.graphType === "treemap") {
+      return this.optionTreeMap;
+    }
+    if (this.graphType === "sankey") {
+      return this.optionSankey;
+    }
+    throw new Error("graphType does not exist");
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  public tooltipFormatter(params: any): string {
```

Comment:
try CallbackDataParams
import { CallbackDataParams } from "echarts/types/dist/shared";

Reply:
@blueur ça fonctionne pas car Property 'treePathInfo' does not exist on type 'CallbackDataParams'.Vetur(2339) et pareil pour value et data. Si t'as une plus belle solution je suis preneur

Reply:
Tu peux aussi étendre CallbackDataParams

Reply:
I tried, and I failed. Thank you echarts